torch/csrc/distributed/c10d/UCCUtils.hpp

```diff
@@ -31,6 +31,33 @@ namespace c10d {
     }                                     \
   } while (0)
 
+// Macro and throw on a non-successful UCC return value and free its request.
+#define TORCH_UCC_CHECK_REQUEST(_request, _cmd, _error_msg) \
+  do {                                                      \
+    ucc_status_t result = _cmd;                             \
+    if (result != UCC_OK) {                                 \
+      std::string err = c10::str(                           \
+          "[",                                              \
+          std::string(__FILE__),                            \
+          ":",                                              \
+          std::to_string(__LINE__),                         \
+          "] ",                                             \
+          logger->getLogPrefix(),                           \
+          _error_msg,                                       \
+          ", error code ",                                  \
+          result,                                           \
+          ": ",                                             \
+          ucc_status_string(result),                        \
+          ", system error code ",                           \
+          errno);                                           \
```

Comment:
This error string preparation looks the same as the one above in `TORCH_UCC_CHECK`.
Do you mind consolidating them? Thanks!